macros/reworked_compare.sql

```diff
@@ -0,0 +1,63 @@
+{% macro reworked_compare(a_query, b_query, primary_key_columns=[], columns=[], event_time=None, sample_limit=20) %}
```

Comment:
There doesn't seem to be any macro called just `compare()` today, so we could maybe use this name instead of `reworked_compare`.

Agree that `reworked_compare` doesn't sound optimal. If we don't want to use `compare()` we should describe what this one does different (optimized, or faster, or whatever) rather than just `reworked`.